pkg/util/overridemanager/labelannotationoverrider.go

```diff
@@ -47,6 +47,10 @@ func buildLabelAnnotationOverriderPatches(rawObj *unstructured.Unstructured, ove
 				continue
 			}
 		}
+		// RFC6902 https://datatracker.ietf.org/doc/html/rfc6902/#appendix-A.14
+		// Refer to a key with ~ or / in its name, you must escape the characters with ~0 and ~1 respectively.
+		// More info: https://jsonpatch.com/
+		key = strings.ReplaceAll(key, "/", "~1")
```

Comment:
Here we just escaped `/`, is there a specific reason not to handle `~`?

Reply:
An example: should `ab/cd~03` be replaced with `ab~1/cd~003`?

Reply:
![image](https://user-images.githubusercontent.com/17962021/211708893-0315ed98-6915-4758-92d3-b2cc7bf76259.png)

 It seems that `~` is not a legal character .

Reply:
What if people set a value with the `~`? like 
```yaml
        annotationsOverrider:
          - operator: add
            value: 
              "testannotation~projectId": "c-m-lfx9lk92:p-v86cf"
```

Reply:
I think we should add a validation rule to make sure both the provided key and value are valid.